### PR TITLE
BreadthFirstSearch in C

### PR DESCRIPTION
**Pull request by kulsoomzahra**
Breadth First Search Algorithm in C.

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)

### Description

(Write your answer here.)

### Checklist

- [ ] I've read the contribution guidelines.
- [ ] I've checked the issue list before deciding what to submit.
- [ ] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(Write your answer here.)

### CODE REVIEW

**Review by siddharth25pandey — APPROVED** [review]
@kulsoomzahra Looks fine to me.
Able to merge.
Thanks

File: C/BreadthFirstSearch.c [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+         printf("Invalid edge!\n");
+         count--;
+      }
+      else
+      {
+         adj[origin][destin] = 1;
+         adj[destin][origin] = 1; // assuming it is a bi-directional graph, we are pushing the reverse edges too.
+      }
+   }
+}
```
Comment (atarax665) [inline C/BreadthFirstSearch.c:125]: Add sample input/output in the end.

File: C/BreadthFirstSearch.c [hunk 2]
```diff
@@ -0,0 +1,125 @@
+BFS Implementation In C:
```
Comment (atarax665) [inline C/BreadthFirstSearch.c:1]: Comment out this line.

File: C/BreadthFirstSearch.c [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+      printf("Queue Overflow\n");
+   else
+   {
+      if(front == -1)
+         front = 0;
+      rear = rear+1;
+      queue[rear] = vertex ;
+   }
+}
+
```
Comment (atarax665) [inline C/BreadthFirstSearch.c:75]: Add docstring to every function for better understanding.

**Review by atarax665 — CHANGES_REQUESTED** [review]

**Review by vatsalkesarwani12 — CHANGES_REQUESTED** [review]
edit readme 
also, select the checklist

**Comment by atarax665** [comment]
Closing this PR as there is no update from contributor's side.

**Pull request closed without merging**